### PR TITLE
ebs_br: allow temporary TiKV unreachable during starting snapshot backup

### LINKED ISSUE #49152: snap_br: snapshot backup may fail if TiKV is temporary unreachable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Make a temporary network isolation between BR and PD.
2. Start snapshot backup.

### 2. What did you expect to see? (Required)
It should success after the network isolation resumed.

### 3. What did you see instead (Required)
BR failed.

### 4. What is your TiDB version? (Required)
Master

### LINKED ISSUE #49153: snap_br: possible inconsistency not reported while TiKV restarts.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Start EBS snapshot backup.
2. During EBS snapshot backup, restart a TiKV node.

### 2. What did you expect to see? (Required)
The backup should fail due to possible consistency break of suspending lightning.

### 3. What did you see instead (Required)
The backup successes.

### 4. What is your TiDB version? (Required)
Master.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #49152, close #49153

Problem Summary:
See the issue.
For #49152, we didn't add retry for starting suspending lightning. 
For #49153, we just break the loop when keeper encounters errors, this may cause the final consistency check passes because of the request of extend lease.

### What changed and how does it work?
Fixed the problems above by retry and fail fast.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fixed a bug that may cause EBS snapshot backup not work properly with TiKV outage.
```

### CODE REVIEW

**Comment by nkg-** [comment]
With this change, what is the maximum time, evict leader scheduler (and other schedulers) can remain paused. If a tikv is already restarted, then init pod will wait for it come up, and then suspend lightning. And during this time, all schedulers/gc/suspend, will be paused right.

**Comment by YuJuncen** [comment]
> With this change, what is the maximum time, evict leader scheduler (and other schedulers) can remain paused. 

If every requests to suspend lightning failed immediately, we will keep retry for about 10 mins. For some call that stuck, we may cost more time over it. If the GC stop time is essential, perhaps we can make the retry based on time cost over failed requests instead of failure count.

> If a tikv is already restarted, then init pod will wait for it come up, and then suspend lightning. And during this time, all schedulers/gc/suspend, will be paused right.

Yes.

**Comment by nkg-** [comment]
> If every requests to suspend lightning failed immediately, we will keep retry for about 10 mins. For some call that stuck, we may cost more time over it. If the GC stop time is essential, perhaps we can make the retry based on time cost over failed requests instead of failure count.

Yeah. Infact, can we make the max pause (gc/schedulers/import) duration configurable. We don't want to pause them more than X (lets say 10 mins). And if during that time, if we cannot pause all tikvs, then its ok to fail the backup. But yeah, retry based on time limit will be ideal. But for implementation, its ok to use retries with exponential backup, and break after a certain time.

A bit outside the scope of this PR. But do we have retries around ebs snapshot trigger (done within backup pod). If create-snapshot api gets throttled, whats the max retry time. Asking since during time, the init pod (and hence pause) will be active. Ok taking this discussion offline (on slack).

**Review by BornChanger — APPROVED** [review]

File: tests/realtikvtest/brietest/operator_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
+// Copyright 2023 PingCAP, Inc. Licensed under Apache-2.0.
+
```
Comment (BornChanger) [inline tests/realtikvtest/brietest/operator_test.go:2]:
```suggestion
```

File: br/pkg/task/operator/cmd.go [hunk 2]
```diff
@@ -130,8 +133,16 @@ func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig) error {
 	return eg.Wait()
 }
 
+func getCallerName() string {
+	name, err := os.Hostname()
```
Comment (BornChanger) [inline br/pkg/task/operator/cmd.go:137]: better add a timestamp in case of from the same client

**Review by 3pointer — APPROVED** [review]

File: br/pkg/task/operator/config.go [hunk 3]
```diff
@@ -14,10 +14,13 @@ type PauseGcConfig struct {
...
 	SafePoint uint64        `json:"safepoint" yaml:"safepoint"`
 	TTL       time.Duration `json:"ttl" yaml:"ttl"`
+
+	OnAllReady func() `json:"-" yaml:"-"`
+	OnExit     func() `json:"-" yaml:"-"`
 }
 
 func DefineFlagsForPrepareSnapBackup(f *pflag.FlagSet) {
-	_ = f.DurationP("ttl", "i", 5*time.Minute, "The time-to-live of the safepoint.")
+	_ = f.DurationP("ttl", "i", 10*time.Minute, "The time-to-live of the safepoint.")
```
Comment (BornChanger) [inline br/pkg/task/operator/config.go:23]:
```suggestion
	_ = f.DurationP("ttl", "i", 2*time.Minute, "The time-to-live of the safepoint.")
```

**Pull request merged**